oeps/oep-0008.rst

```diff
@@ -0,0 +1,374 @@
+  maintainers themselves suggest using nose2 or pytest (or plain
+  unittest/unittest2). As we still want to use a framework that extends
+  unittest, the question then becomes whether edX standardize on nose2 or
+  pytest.
+
+  Note that edX developers have already begun using pytest in new repos,
+  having chosen it over nose2 for some or all of the reasons listed below.
+  Given this and lacking other barriers that would make it a bad choice, we
+  have decided to standardize on pytest.
+
+  Here are some of the reasons that we have chosen pytest:
+
+  * Actively maintained and well documented
+
+  * Industry momentum - pytest has garnered widespread adoption, and thus
+    most current tutorials, templates, blog posts, etc. provide examples
+    that use this framework
+
+  * Highly customizable, a good API, a plugin system (both included and
+    community plugins)
+
+  * Ability to mark tests, and to programmatically select which tests to run
+    by name, path, module, or mark
+
+  * Better, more customizable failure reports
+
+  * Explicit, modular, scalable fixtures
+
+  * Built-in features such as a --duration flag to show the slowest tests,
+    console output capture by default, traceback modification
+
+  * Can run current code: unittest, nose, doctest
+
+`rake`_ / `Invoke`_
+  The original choice in the edx-platform was to use rake as the task
+  manager. However that made little sense as the amount of Ruby code in
+  the platform is eclipsed by the amount of Python code, and most
+  developers are much more comfortable with Python.
+
+  Paver was chosen over Invoke (pyinvoke) for reasons including the
+  following:
+
+  * While fully usable, Invoke is still pre-1.0 software and has no
+    backwards compatibility guarantees until the 1.0 release occurs.
+
+  * Paver is used as the task management / scripting tool for the edx-
+    platform codebase, and we have good examples there of its usage,
+    including tests.
+
+`JSHint`_ / `JSCS`_
+  The original choice in the edx-platform was to use JSHint for JavaScript code
+  linting.
+
+  However JSHint didn't have a good story for linting ES2015+ as we aimed to
+  upgrade to that version of JavaScript.
+
+  And JSCS, which was another leader in the JS code style linter and formatter
+  has merged with ESLint.
+
+  Thus we determined that ESLint is preferred to cover our immediate, short
+  term, and long term needs.
+
+`Coveralls`_
+  The original choice for recording code coverage for trending
+  reports was to write the coverage statistics from edx-platform test
+  execution to DataDog.
+
+  This homegrown method was neither scalable no maintainable. As such, we
+  started integrating with Coveralls.
+
+  However the SLA for Coveralls posting the data back to GitHub does not meet
+  our needs, and developers were waiting many hours after the test execution
+  on the CI system had completed to get the feedback on their PRs. This led to
```

Comment:
I think it's important to include at least some reasoning so future us knows why we made the decision. However, that reasoning might be "it seemed like a good idea at the time"